cmd/otel-agent/config/agent_config.go

```diff
@@ -0,0 +1,122 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package config provides a way to convert the OpenTelemetry Collector configuration to the Datadog Agent configuration.
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/datadogexporter"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/confmap/provider/envprovider"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
+	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/service"
+)
+
+// NewConfigComponent creates a new config component from the given URIs
+func NewConfigComponent(ctx context.Context, uris []string) (config.Component, error) {
+	// Load the configuration from the fileName
+	rs := confmap.ResolverSettings{
+		URIs: uris,
+		ProviderFactories: []confmap.ProviderFactory{
+			fileprovider.NewFactory(),
```

Comment:
To make sure I understand how this works: are these providers adding functionality that control how the configuration is created? Or are they stubs to be filled in later? If the former, how are they themselves configured? That is, what controls the file used for configuration, or the http / https endpoint used for configuration, etc?

Reply:
Once the file location follows https://datatracker.ietf.org/doc/html/rfc8089#section-2, the providers are pre configured to handle it . see 
 https://github.com/open-telemetry/opentelemetry-collector/tree/main/confmap#provider